game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java

```diff
@@ -113,8 +114,8 @@ private void addChat() {
     remove(chatSplit);
     chatPanelHolder.removeAll();
     final ChatModel chat = chatPanelSupplier.get().orElse(null);
-    if (chat instanceof SetupPanel) {
-      chatPanelHolder.add((SetupPanel) chat, BorderLayout.CENTER);
+    if (chat instanceof ChatPanel) {
```

Comment:
If someone were to ever replace `ChatPanel` with a new subclass, but one that still maintained the same contract and woudl work, this instanceof would break. That is particularly bad to consider a change in some other, far-away, location then broke code here. 

`instanceof`:
- generally considered to be brittle
- often indicates poor design and abstractions.
- Using type information for control flow is essentially attaching an undeclared boolean flag to the object type which the 'caller' code then owns. For example, instead of having a headless flag every 'caller' could do an `instanceof` check and decide for itself what the definition of 'headless' is. It's a significant violation of data encapsulation.


IMO the original bug here was in part because an inheritently`instanceof` check is used. From my view instanceof is best used where strictly necessary and is something you tend to want to reduce to zero.

Also because as mentioned, if the sub-class were changes this would break. AFAIK the latest best practice advice is to favor using [isAssignableFrom](https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#isAssignableFrom(java.lang.Class)) instead.